File: src/main/java/com/soapdogg/lispInterpreter/generator/TokenGenerator.java
package com.soapdogg.lispInterpreter.generator;

import com.soapdogg.lispInterpreter.datamodels.Token;
import com.soapdogg.lispInterpreter.datamodels.TokenKind;
import com.soapdogg.lispInterpreter.constants.TokenValueConstants;
import lombok.AllArgsConstructor;

@AllArgsConstructor(staticName = "newInstance")
public class TokenGenerator {

    public Token generateCloseToken() {
        return new Token(
            TokenKind.CLOSE_TOKEN,
            String.valueOf(TokenValueConstants.CLOSE_PARENTHESES)
        );
    }

    public Token generateOpenToken() {
        return new Token(
            TokenKind.OPEN_TOKEN,
            String.valueOf(TokenValueConstants.OPEN_PARENTHESES)
        );
    }

    public Token generateNumericToken(
        String value
    ) {
        return new Token(
            TokenKind.NUMERIC_TOKEN,
            value
        );
    }

    public Token generateLiteralToken(
        String value
    ) {
        return new Token(
            TokenKind.LITERAL_TOKEN,
            value
        );
    }
}

File: src/main/java/com/soapdogg/lispInterpreter/generator/TokenGenerator.kt
package com.soapdogg.lispInterpreter.generator

import com.soapdogg.lispInterpreter.constants.TokenValueConstants
import com.soapdogg.lispInterpreter.datamodels.Token
import com.soapdogg.lispInterpreter.datamodels.TokenKind

class TokenGenerator {
    fun generateCloseToken(): Token {
        return Token(
            TokenKind.CLOSE_TOKEN,
            TokenValueConstants.CLOSE_PARENTHESES.toString()
        )
    }

    fun generateOpenToken(): Token {
        return Token(
            TokenKind.OPEN_TOKEN,
            TokenValueConstants.OPEN_PARENTHESES.toString()
        )
    }

    fun generateNumericToken(
        value: String
    ): Token {
        return Token(
            TokenKind.NUMERIC_TOKEN,
            value
        )
    }

    fun generateLiteralToken(
        value: String
    ): Token {
        return Token(
            TokenKind.LITERAL_TOKEN,
            value
        )
    }
}